code/modules/mob/living/carbon/xenomorph/castes/baneling/baneling.dm

```diff
@@ -0,0 +1,72 @@
+/mob/living/carbon/xenomorph/baneling
+	caste_base_type = /mob/living/carbon/xenomorph/baneling
+	name = "Baneling"
+	desc = "A green alien that store chemicals and can run pretty fast . . ."
+	icon = 'icons/Xeno/2x2_Xenos.dmi'
+	icon_state = "Baneling Walking"
+	bubble_icon = "alienleft"
+	health = 100
+	maxHealth = 100
+	plasma_stored = 50
+	tier = XENO_TIER_ONE
+	upgrade = XENO_UPGRADE_ZERO
+	pixel_x = -16
+	old_x = -16
+
+/obj/structure/xeno/baneling_pod
+	name = "Baneling Pod"
+	desc = "A baneling pod, storing fresh banelings "
+	icon = 'icons/Xeno/2x2_Xenos.dmi'
+	icon_state = "Baneling Pod"
+	density = FALSE
+	obj_flags = CAN_BE_HIT | PROJ_IGNORE_DENSITY
+	/// Ref to our xeno
+	var/mob/living/carbon/xenomorph/xeno_ref
+
+/obj/structure/xeno/baneling_pod/Initialize(mapload, _hivenumber, xeno, ability_ref)
+	. = ..()
+	xeno_ref = xeno
+	RegisterSignal(xeno_ref, COMSIG_MOB_DEATH, PROC_REF(handle_baneling_death))
+	RegisterSignal(xeno_ref, COMSIG_QDELETING, PROC_REF(obj_destruction))
+	RegisterSignal(ability_ref, COMSIG_ACTION_TRIGGER, PROC_REF(obj_destruction))
+	addtimer(CALLBACK(src, PROC_REF(increase_charge)), BANELING_CHARGE_GAIN_TIME)
+
+/obj/structure/xeno/baneling_pod/obj_destruction()
+	if(length(contents) >= 1)
```

Comment:
This block only runs when there's a xeno inside, and the block below only runs when the xeno is dead. When's the xeno dead but not in the pod?

Reply:
When the pod is dead